app/controllers/AuthenticationController.scala

```diff
@@ -274,14 +266,12 @@ class AuthenticationController @Inject()(
     foundFox.futureBox.map(_.isDefined)
   }
 
-  private def canAccessAnnotation(user: User,
-                                  ctx: DBAccessContext,
-                                  annotationTyp: String,
-                                  annotationId: String): Fox[Boolean] = {
+  private def canAccessAnnotation(user: User, ctx: DBAccessContext, annotationId: String): Fox[Boolean] = {
     val foundFox = for {
-      annotation <- annotationProvider.provideAnnotation(annotationTyp, annotationId, user)(ctx)
+      annotationIdParsed <- ObjectId.parse(annotationId)
+      annotation <- annotationDAO.findOne(annotationIdParsed)(GlobalAccessContext)
       _ <- bool2Fox(annotation.state != Cancelled)
-      restrictions <- annotationProvider.restrictionsFor(annotationTyp, annotationId)(ctx)
+      restrictions <- annotationProvider.restrictionsFor(AnnotationIdentifier(annotation.typ, annotationIdParsed))(ctx)
```

Comment:
annotationDAO.findOne will always either return a Task, Explorational, or nothing. Compound annotations cannot be retreived in this way. So either there is an annotation (which will have a valid `typ` (either `Task` or `Explorational) or the findOne already fails

Reply:
Cool! Just out of curiosity: what's the reason to wrap the arguments in the `AnnotationIdentifier`?

Reply:
Since we now already have the enum and ObjectId (fields on the annotation), we can use this faster version of restrictionsFor (the prevously used (String, String) one first parses the individual components).
I’d say grouping the two things in the AnnotationIdentifier case class is a bit unnecessary, but that’s currently what the annotationProvider wants. So you could say historical reasons.